stdlib/src/base64/_b64encode.mojo

```diff
@@ -0,0 +1,408 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+"""
+We make use of the following papers for the implementation, note that there
+are some small differences.
+Wojciech Muła, Daniel Lemire, Base64 encoding and decoding at almost the
+speed of a memory copy, Software: Practice and Experience 50 (2), 2020.
+https://arxiv.org/abs/1910.05109
+
+Wojciech Muła, Daniel Lemire, Faster Base64 Encoding and Decoding using AVX2
+Instructions, ACM Transactions on the Web 12 (3), 2018.
+https://arxiv.org/abs/1704.00605
+"""
+
+from collections import InlineArray
+from memory import memcpy, bitcast, UnsafePointer
+from memory.maybe_uninitialized import UnsafeMaybeUninitialized
+
+
+fn _subtract_with_saturation[
+    simd_size: Int, //, b: Int
+](a: SIMD[DType.uint8, simd_size]) -> SIMD[DType.uint8, simd_size]:
+    """The equivalent of https://doc.rust-lang.org/core/arch/x86_64/fn._mm_subs_epu8.html .
+    This can be a single instruction on some architectures.
+    """
+    alias b_as_vector = SIMD[DType.uint8, simd_size](b)
+    return max(a, b_as_vector) - b_as_vector
+
+
+"""
+| 6-bit Value | ASCII Range | Target index | Offset (6-bit to ASCII) |
+|-------------|-------------|--------------|-------------------------|
+| 0 ... 25    | A ... Z     | 13           | 65                      |
+| 26 ... 51   | a ... z     | 0            | 71                      |
+| 52 ... 61   | 0 ... 9     | 1 ... 10     | -4                      |
+| 62          | +           | 11           | -19                     |
+| 63          | /           | 12           | -16                     |
+"""
+alias UNUSED = 0
+# fmt: off
+alias TABLE_BASE64_OFFSETS = SIMD[DType.uint8, 16](
+    71,                                     # a ... z
+    -4, -4, -4, -4, -4, -4, -4, -4, -4, -4, # 0 ... 9 
+    -19,                                    # +
+    -16,                                    # /
+    65,                                     # A ... Z
+    UNUSED, UNUSED
+)
+# fmt: on
+alias END_FIRST_RANGE = 25
+alias END_SECOND_RANGE = 51
+
+
+fn _get_simd_range_values[simd_width: Int]() -> SIMD[DType.uint8, simd_width]:
```

Comment:
Nit: add a FIXME indicating that this function is introduced because `math.iota` doesn't run at compile time. Maybe even consider renaming to `_iota`.

Reply:
Good call. I didn't know we had a function for this. Probably because of how cryptic the name `iota` is (not everyone is a greek fan). I moved the function next to `iota` and added a `TODO`.